src/bin/lychee/main.rs

```diff
@@ -160,9 +183,33 @@ async fn run(cfg: &Config, inputs: Vec<Input>) -> Result<i32> {
         clients.listen().await;
     });
 
-    while let Some(response) = recv_resp.recv().await {
+    let input_domains: HashSet<String> = input_domains(inputs);
+
+    // We keep track of the total number of requests
+    // and exit the loop once we reach it.
+    // Otherwise the sender would never be dropped and
+    // we'd be stuck indefinitely.
+    let mut curr = 0;
+
+    while curr < total_requests {
+        curr += 1;
```

Comment:
@lebensterben, I'm not too happy about the above three lines. Is there a better way to handle the request queue compared to counting the number of total requests like I did?

Reply:
Check this
https://docs.rs/tokio-stream/0.1.5/tokio_stream/#returning-a-stream-from-a-function
more examples:
https://tokio.rs/tokio/tutorial/streams

Reply:
tokio-stream crate has a wrapper type for Reciever of the channel.
That should work without counting the number.
It's like Future trait, you can poll it.
It's also like Iterator trait, `next` returns None when it's over.

Reply:
Ah that's great. Sounds like what we need.